### PR TITLE
Use RFC5424 / RFC5426 syslog over UDP format

### PR DESCRIPTION
**Pull request by bgentry**
Also adds timestamp and hostname options.

Note that this is likely backwards-incompatible for most users unless they change their syslog-ng configs. We should probably bump the version for it.

### CODE REVIEW

**Comment by bgentry** [comment]
I'm wondering if the switch from RFC3164 to RFC5424 format will cause any issues for other users of this module.

**Comment by jkvor** [comment]
:+1:

will need to pull this commit into the `production` branch as well

**Comment by bgentry** [comment]
@JacobVorreuter let's wait until I verify with tcpdump that I'm actually writing the correct thing to the wire. I'm not getting the expected results, but that could also be due to incorrect syslog configs.

File: README.md [hunk 1]
```diff
@@ -1,7 +1,15 @@
+## Overview
+
+This library writes messages formatted as per [RFC5424][rfc5424], transmitted
+via UDP as described in [RFC5426][rfc5426]. Expect this format on port 514,
+which hermes sends userlogs to. Also, respect the timestamps coming from these
```
Comment (jkvor) [inline README.md]: probably shouldn't reference hermes in docs for an open-source dependency

**Comment by jkvor** [comment]
It's worth noting that the `production` branch has a single `gen_server:cast` on the critical logging path.  This branch adds a `gen_server:call` for looking up the `facility` and potentially another `gen_server:call` for looking up the `app_name`.  Could these create bottlenecks in a high-throughput logging environment?

File: README.md (hunk 1 above)
Comment (bgentry) [inline README.md]: yeah that line wasn't meant for the readme. fixed.

**Comment by jkvor** [comment]
:+1: to merge

**Pull request merged**